src/main/docs/guide/aop/aroundAdvice.adoc

```diff
@@ -72,37 +51,13 @@ The semantics of AOP advice when applied to <<factories,Bean Factories>> differs
 
 Consider the following two examples:
 
-.AOP Advice at the type level of a `@Factory`
-[source,java]
-----
-@Cacheable("my-cache")
-@Factory
-class MyFactory {
-
-    @Prototype
-    MyBean myBean() {
-        return new MyBean();
-    }
-}
-----
+snippet::io.micronaut.docs.aop.advice.type.MyFactory[tags="class", indent=0, title="AOP Advice at the type level of a `@Factory`"]
 
 The above example will cache creation of `MyBean` bean, according to the semantics defined by the ann:cache.annotation.Cacheable[] cache configuration.
 
 Now consider this example:
 
-.AOP Advice at the method level of a `@Factory`
-[source,java]
-----
-@Factory
-class MyFactory {
-
-    @Prototype
-    @Cacheable("my-cache")
-    MyBean myBean() {
-        return new MyBean();
-    }
-}
-----
+snippet::io.micronaut.docs.aop.advice.method.MyFactory[tags="class", indent=0, title="AOP Advice at the method level of a `@Factory`"]
 
 The above example will cache all of the public methods of the `MyBean` bean, but not the bean creation.
 
```

Comment:
Is there a way to clean up the language of the end of the paragraphs below? I'm not even sure what it means so I can't be the only one reading it who doesn't, but that I also can't clarify it without understanding it.